src/server/scripts/Spells/spell_shaman.cpp

```diff
@@ -456,6 +462,110 @@ class spell_sha_earthen_rage_proc_aura : public AuraScript
     }
 };
 
+// 61882 - Earthquake
+class spell_sha_earthquake : public SpellScript
+{
+    PrepareSpellScript(spell_sha_earthquake);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_SHAMAN_AFTERSHOCK, SPELL_SHAMAN_AFTERSHOCK_ENERGIZE });
+    }
+
+    void HandleAfterCast()
+    {
+        int32 energize = *GetSpell()->GetPowerTypeCostAmount(POWER_MAELSTROM);
+
+        if (AuraEffect* aftershock = GetCaster()->GetAuraEffect(SPELL_SHAMAN_AFTERSHOCK, EFFECT_0))
+            if (roll_chance_f(aftershock->GetAmount()))
+                GetCaster()->CastSpell(GetCaster(), SPELL_SHAMAN_AFTERSHOCK_ENERGIZE, CastSpellExtraArgs(energize)
+                    .AddSpellBP0(energize));
+    }
+
+    void Register() override
+    {
+        AfterCast += SpellCastFn(spell_sha_earthquake::HandleAfterCast);
+    }
+};
+
+// 61882 - Earthquake aura
+class spell_sha_earthquake_aura : public AuraScript
+{
+    PrepareAuraScript(spell_sha_earthquake_aura);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_SHAMAN_EARTHQUAKE, SPELL_SHAMAN_EARTHQUAKE_TICK });
+    }
+
+    void HandlePeriodic(AuraEffect const* /*aurEff*/)
+    {
+        std::vector<AreaTrigger*> areaTriggers = GetCaster()->GetAreaTriggers(SPELL_SHAMAN_EARTHQUAKE);
+
+        for (AreaTrigger* at : areaTriggers)
+            GetTarget()->CastSpell(at->GetPosition(), SPELL_SHAMAN_EARTHQUAKE_TICK, true);
+    }
+
+    void Register() override
+    {
+        OnEffectPeriodic += AuraEffectPeriodicFn(spell_sha_earthquake_aura::HandlePeriodic, EFFECT_1, SPELL_AURA_PERIODIC_DUMMY);
+    }
+};
+
+// 77478 - Earthquake tick
+class spell_sha_earthquake_tick : public SpellScript
+{
+    PrepareSpellScript(spell_sha_earthquake_tick);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo
+        ({
+            SPELL_SHAMAN_ELEMENTAL_SHAMAN,
+            SPELL_SHAMAN_BOTTLE_OF_SWIRLING_MAELSTROM,
+            SPELL_SHAMAN_EARTHQUAKE_KNOCKING_DOWN
+        });
+    }
+
+    void CalcDamage(SpellEffIndex /*effIndex*/)
+    {
+        const Player* player = GetCaster()->ToPlayer();
```

Comment:
possible crash: make sure to check if player is null if you want to cast caster to it, imagine some gm letting npcs cast this spell forcefully to crash the server